### ISSUE TITLE
Cross-validation split configuration pane in the UI

### ISSUE THREAD

**Issue by aymuos15** [aymuos15/AutoMONAI#27]
## Problem

The `--cross_val K` and `--cv_fold N` CLI flags exist and the UI generates fold variants with per-fold state tracking (#15), but **`run.py` ignores these flags entirely**. Every fold trains on the exact same full `imagesTr/` dataset — there is no actual train/val split. The fold system is currently UI scaffolding only.

## Current behaviour

- `get_train_files()` returns **all** files from `imagesTr/` — used as the full training set
- `imagesTs/` is only used during `--mode infer`
- **No validation set** exists during training — loss is computed on training data only
- `--cross_val` and `--cv_fold` are parsed but never read by `run.py`

### 1. Data splitting in core (the critical missing piece)

Add to `dataset.py`:
```python
def split_train_val(files, mode="none", val_ratio=0.2, n_folds=5, fold=None, seed=42):
    """Split training files into train/val sets.
    
    Returns (train_files, val_files).
    """
```

Wire into `run.py` so that `--cross_val K --cv_fold N` actually holds out fold N as validation and trains on the remaining K-1 folds.

### 2. Split options

| Mode | Description |
|---|---|
| **No validation** (current default) | All `imagesTr/` used for training, no val set |
| **Hold-out split** | Fixed train/val split by percentage (e.g. 80/20) |
| **K-fold CV** | K-fold split — fold N is val, rest is train |
| **Custom split file** | JSON file mapping filenames to fold indices |

### 3. UI controls (Cross-Validation pane in Generate tab)
- **Split mode** dropdown: `none` / `holdout` / `kfold` / `custom`
- **Holdout ratio** slider (0.1–0.5, default 0.2) — visible when mode = `holdout`
- **Fold count** input (2–20, default 5) — visible when mode = `kfold`
- **Split seed** input (default: 42) — for reproducibility
- **Custom split file** path input — visible when mode = `custom`
- Preview showing train/val sample counts for the selected configuration

### 4. CLI flags
```
--val_split         none | holdout | kfold | custom (default: none)
--val_ratio         Hold-out validation ratio (default: 0.2)
--cross_val K       Number of folds (existing flag, default: 5)
--cv_fold N         Run only fold N (existing flag)
--split_seed        Random seed for splitting (default: 42)
--split_file        Path to custom split JSON
```

### 5. Training loop changes
- Create validation DataLoader from `val_files`
- Compute val metrics each epoch (val loss, val Dice, etc.)
- Log val metrics to W&B with `val/` prefix
- Use val loss for early stopping and best model selection (instead of train loss)
- Save `split.json` to run directory for reproducibility on resume

### 6. Command output
```
python3 -m automonai.core.run --dataset ... --val_split kfold --cross_val 5 --cv_fold 1 --split_seed 42
```
